service/history/workflow/mutable_state_impl.go

```diff
@@ -7387,3 +7394,19 @@ func (ms *MutableStateImpl) reschedulePendingActivities() error {
 
 	return nil
 }
+
+func (ms *MutableStateImpl) AddReapplyCandidateEvent(event *historypb.HistoryEvent) {
+	switch event.GetEventType() {
+	case enumspb.EVENT_TYPE_WORKFLOW_EXECUTION_SIGNALED,
```

Comment:
I guess nexus handles in a different path?

Reply:
I don't see nexus handling it: https://github.com/temporalio/temporal/blob/b44aae88e0aa255b8dc4fc6ffb1735b8f7b3519b/service/history/ndc/hsm_state_replicator.go#L140 may be looking at wrong place.

Reply:
syncHSM task is only used for state-only changes and will confirm event version history is update to date before syncing so doesn't need to handle the reapply case.

Here we need to handle nexus events as well.

Reply:
updated according to offline sync.